Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -56,6 +56,13 @@ public File getCacheFile(ParseFile.State state) {
     return new File(cachePath, state.name());
   }
 
+  /* package for tests */ File getTempFile(ParseFile.State state) {
+    if (state.url() == null) {
+      return null;
+    }
+    return new File(cachePath, ParseDigestUtils.md5(state.url()));
```

Comment:
The current implementation is fine, but why not just use state.url + (.tmp) or something? We should also remove all temp files for a given file that exist at some point, maybe when we try to re-download?